rust/rsc/src/rsc/main.rs

```diff
@@ -44,30 +49,51 @@ struct ServerOptions {
 
     #[arg(help = "Launches the cache without an external database", long)]
     standalone: bool,
+
+    #[arg(
+        help = "Specify an absolute path for a local directory based store",
+        value_name = "LOCAL_STORE",
+        long
+    )]
+    local_store: Option<String>,
 }
 
-fn create_router(state: Arc<DatabaseConnection>) -> Router {
+fn create_router(conn: Arc<DatabaseConnection>, config: Arc<config::RSCConfig>) -> Router {
     Router::new()
         .route(
             "/job",
             post({
-                let shared_state = state.clone();
-                move |body| add_job::add_job(body, shared_state)
+                let conn = conn.clone();
+                move |body| add_job::add_job(body, conn)
             })
             .layer(axum::middleware::from_fn({
-                let shared_state = state.clone();
-                move |req, next| {
-                    api_key_check::api_key_check_middleware(req, next, shared_state.clone())
-                }
+                let conn = conn.clone();
+                move |req, next| api_key_check::api_key_check_middleware(req, next, conn.clone())
             })),
         )
         .route(
             "/job/matching",
             post({
-                let shared_state = state.clone();
-                move |body| read_job::read_job(body, shared_state)
+                let conn = conn.clone();
+                move |body| read_job::read_job(body, conn)
+            }),
+        )
+        .route(
```

Comment:
Add a todo here about using the blob store for the url retrieval.

Reply:
The upload url is not really tied to the blob store. At least for as long as we do the double upload. We mostly kept this route as a safety net for ourselves but tbh I think each blob store will be so different that it won't ever make since to change this.